### PR TITLE
Braidingtensor improvements

### PR DESCRIPTION
**Pull request by lkdvos**
This PR fixes some things about BraidingTensor that were either not working or just wrong, along with fixing the whole `braid(::AbstractTensorMap)` part of the code since apparently that was just not working.
It also adds some tests to catch this better in the future

### CODE REVIEW

File: src/tensors/braidingtensor.jl [hunk 1]
```diff
@@ -104,25 +101,12 @@ end
...
-            a1, a2 = f₂.uncoupled
-            c = f₂.coupled
-            f₁.uncoupled == (a2, a1) || continue
-            braiddict = artin_braid(f₂, 1; inv=b.adjoint)
-            r = convert(scalartype(t), get(braiddict, f₁, zero(valtype(braiddict))))
-        end
-        @inbounds for i in axes(data, 1), j in axes(data, 2)
-            data[i, j, j, i] = r
-        end
+function Base.copy!(tdst::TensorMap, tsrc::BraidingTensor)
```
Comment (Jutho) [inline src/tensors/braidingtensor.jl]: This is probably equivalent to the generic `copy!(::AbstractTensorMap,::AbstractTensorMap)`

File: test/planar.jl [hunk 2]
```diff
@@ -30,22 +30,59 @@ function force_planar(tsrc::TensorMap{<:Any,<:GradedSpace})
...
-    t2 = @constinferred BraidingTensor{ComplexF64}(V1)
-    @test scalartype(t2) == ComplexF64
-    @test storagetype(t2) == Vector{ComplexF64}
-
-    V2 = ℂ^2 ⊗ ℂ^3 ← ℂ^2 ⊗ ℂ^3
-    @test_throws SpaceMismatch BraidingTensor(V2)
-
-    @test adjoint(t1) isa BraidingTensor
+    for V in (Vtr, VU₁, VfU₁, VfSU₂, Vfib)
+        V1 = V[1] ⊗ V[2] ← V[2] ⊗ V[1]
```
Comment (Jutho) [inline test/planar.jl]: I guess we coud use `W` for the `HomSpace` object, to avoid confusion with `V[1]`.

**Review by Jutho — APPROVED** [review]

File: test/planar.jl [hunk 3]
```diff
@@ -30,22 +30,59 @@ function force_planar(tsrc::TensorMap{<:Any,<:GradedSpace})
...
+        @test space(t1) == W
+        @test codomain(t1) == codomain(W)
+        @test domain(t1) == domain(W)
+        @test scalartype(t1) == (isreal(sectortype(W)) ? Float64 : ComplexF64)
+        @test storagetype(t1) == Vector{scalartype(t1)}
+        t2 = @constinferred BraidingTensor{ComplexF64}(W)
+        @test scalartype(t2) == ComplexF64
+        @test storagetype(t2) == Vector{ComplexF64}
+
+        V2 = reverse(codomain(W)) ← domain(V1)
```
Comment (lkdvos) [inline test/planar.jl]:
```suggestion
        W2 = reverse(codomain(W)) ← domain(W)
```

File: test/planar.jl [hunk 4]
```diff
@@ -30,22 +30,59 @@ function force_planar(tsrc::TensorMap{<:Any,<:GradedSpace})
...
+        @test_throws SpaceMismatch BraidingTensor(V2)
```
Comment (lkdvos) [inline test/planar.jl]:
```suggestion
        @test_throws SpaceMismatch BraidingTensor(W2)
```

**Comment by Jutho** [comment]
It seems like you managed to break LRUCache?

**Comment by lkdvos** [comment]
This should be ready to go

**Pull request merged**